pkg/provisioner/controller.go

```diff
@@ -606,25 +606,25 @@ func (c *obcController) deleteResources(ob *v1alpha1.ObjectBucket, cm *corev1.Co
 }
 
 // Add finalizer and labels to the OBC.
-func (c *obcController) setOBCMetaFields(obc *v1alpha1.ObjectBucketClaim) (err error) {
+func (c *obcController) setOBCMetaFields(obc *v1alpha1.ObjectBucketClaim) (*v1alpha1.ObjectBucketClaim, error) {
 	clib := c.libClientset
 
 	logD.Info("getting OBC to set metadata fields")
-	obc, err = clib.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(context.TODO(), obc.Name, metav1.GetOptions{})
+	obcFetched, err := clib.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(context.TODO(), obc.Name, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("error getting obc: %v", err)
+		return nil, fmt.Errorf("error getting obc: %v", err)
```

Comment:
Even in an error, it could be good to return the original OBC itself, just in case code that follows needs to process it in an error state or in case the code in a future time attempts to continue on after an error.

```suggestion
		return obc, fmt.Errorf("error getting obc: %v", err)
```